src/LinAlg/CMakeLists.txt

```
@@ -46,17 +46,52 @@ if(HIOP_WITH_KRON_REDUCTION)
   set(hiopLinAlg_SRC ${hiopLinAlg_SRC} hiopLinSolverUMFPACKZ.cpp)
 endif()
 
-# Build and link linear algebra target
+# Create legacy library
 add_library(hiopLinAlg OBJECT ${hiopLinAlg_SRC})
 target_link_libraries(hiopLinAlg PUBLIC hiop_math)
-
-# Link linear algebra target to Umpire & RAJA when enabled
 if(HIOP_USE_RAJA)
   target_link_libraries(hiopLinAlg PUBLIC umpire RAJA OpenMP::OpenMP_CXX)
 endif()
+# ---
+
+set(hiopLinAlg_HEADERS
+  hiopLinAlgFactory.hpp
+  hiopLinSolver.hpp
+  hiopLinSolverIndefDenseLapack.hpp
```

```suggestion
set(hiopLinAlg_HEADERS
  hiopLinSolver.hpp
  hiopLinSolverIndefDenseLapack.hpp
```
I believe we should not have linear algebra factory headers in HiOp API. These should be used internally within HiOp, but not exposed to applications linking to HiOp.

---

Great catch!